### ISSUE TITLE
FabricJS loadFromJSON() - is there any way to catch Error ?

### ISSUE THREAD

**Issue by xiyu714** [fabricjs/fabric.js#6286]
## Version
3.6.0

## Test Case
http://jsfiddle.net/mbv2ft7x/2/

---
The image src in JSON may have been deleted or is the wrong one.
In the console, display this error "Error loading https://test.jpeg/". Can I catch an error like this? and get the error message.

**Comment by asturur** [comment]
not really, at least not for now.
This would be something that is useful in an API and we should add it.

**Comment by xiyu714** [comment]
@asturur Thanks for the reply.
I tried to override the fabric.log() method to catch the error.
It seems to work.

demo: http://jsfiddle.net/ta72r0bh/3/

**Comment by Niels-NTG** [comment]
> not really, at least not for now.
> This would be something that is useful in an API and we should add it.

In the world of modern JavaScript it would make a lot of sense for `canvas.loadFromJSON` to return a Promise, which you can use to call a callback when it resolves successfully or catch an error if needed. The reviver part should probably stay the same as it is now. In practise this could look something like this:
```js
// OLD: canvas.loadFromJSON(json, callback, reviver)
// Proposal:
canvas.loadFromJSON(json, reviver).then(callback).catch(optionalErrorHandlingCallback)
```

Currently, the only way to know IF something went wrong is by checking if `canvas.loadFromJSON` returned `undefined` instead of the canvas instance. But then it's still unclear what went wrong exactly.

**Comment by asturur** [comment]
Yes i think the next update should remove the callbacks in favor of promises.

**Comment by ShaMan123** [comment]
closed by #3684
